src/ui/public/vis/vis.js

```diff
@@ -125,5 +126,24 @@ export default function VisFactory(Notifier, Private) {
     });
   };
 
+  Vis.prototype.hasUiState = function () {
+    return !!this.__uiState;
+  };
+  Vis.prototype.setUiState = function (uiState) {
+    this.__uiState = uiState;
+  };
+  Vis.prototype.getUiState = function () {
+    return this.__uiState;
+  };
+  Vis.prototype.uiStateVal = function (key, val) {
+    if (this.hasUiState()) {
+      if (_.isUndefined(val)) {
+        return this.__uiState.get(key);
```

Comment:
You never check that `this.__uiState` is actually a persisted state thing, so you can pass in anything, including a raw object, and this will fail, as there is no native `get` method.